detect_secrets/transformers/yaml.py

```diff
@@ -210,25 +210,28 @@ def _compose_node_shim(
         self,
         parent: Optional[yaml.nodes.Node],
         index: Optional[yaml.nodes.Node],
-    ) -> yaml.nodes.Node:
+    ) -> Optional[yaml.nodes.Node]:
         line = (
             self.loader.marks[-1].line
             if self.is_inline_flow_mapping_key
             else self.loader.line
         )
 
-        node = yaml.composer.Composer.compose_node(self.loader, parent, index)
-        node.__line__ = line + 1
+        node = yaml.composer.Composer.compose_node(self.loader, parent, index)  # type: ignore
+        if node is None:
+            return None
+
+        node.__line__ = line + 1    # type: ignore
```

Comment:
Could you please expand on these changes? I'm unsure on what you're trying to achieve here.

Reply:
basically the yaml stubs suggest a return value of either `Node | None`, therefore I wanted to make sure it is not `None`.